tests/ExecutorTest.php

```diff
@@ -356,6 +397,70 @@ public function provideScenarios(): array
                     $this->assertEmpty($response['body']['errors']);
                 }
             ],
+            [
+                'image' => 'openruntimes/node:v4-21.0',
+                'entrypoint' => 'index.js',
+                'folder' => 'node-binary-response',
+                'version' => 'v4',
+                'startCommand' => 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "pm2 start src/server.js --no-daemon"',
+                'buildCommand' => 'tar -zxf /tmp/code.tar.gz -C /mnt/code && helpers/build.sh "npm i"',
+                'assertions' => function ($response) {
+                    $this->assertEquals(200, $response['headers']['status-code']);
+                    $this->assertEquals(200, $response['body']['statusCode']);
+                    $bytes = unpack('C*byte', $response['body']['body']);
+                    if (!is_array($bytes)) {
+                        $bytes = [];
+                    }
+                    self::assertCount(3, $bytes);
+                    self::assertEquals(0, $bytes['byte1'] ?? 0);
+                    self::assertEquals(10, $bytes['byte2'] ?? 0);
+                    self::assertEquals(255, $bytes['byte3'] ?? 0);
+                    $this->assertEmpty($response['body']['logs']);
+                    $this->assertEmpty($response['body']['errors']);
+                },
+                null, // body
+                'mimeType' => 'multipart/form-data'
+            ],
+            [
+                'image' => 'openruntimes/node:v4-21.0',
+                'entrypoint' => 'index.js',
+                'folder' => 'node-binary-response',
+                'version' => 'v4',
+                'startCommand' => 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "pm2 start src/server.js --no-daemon"',
+                'buildCommand' => 'tar -zxf /tmp/code.tar.gz -C /mnt/code && helpers/build.sh "npm i"',
+                'assertions' => function ($response) {
+                    $this->assertEquals(400, $response['headers']['status-code']);
+                    $this->assertStringContainsString("JSON response does not allow binaries", $response['body']['message']);
+                },
+                null, // body
+                'mimeType' => 'application/json'
+            ],
+            [
+                'image' => 'openruntimes/node:v4-21.0',
```

Comment:
Do we also need a test for `node-binary-request` for `application/json` mime type?

Reply:
Binary request wont work with JSON format. JSON property cannot be a binary.